DirectX11Proj/textureshaderclass.cpp

```diff
@@ -0,0 +1,318 @@
+#include "textureshaderclass.h"
+#include "ShaderHelperFunctions.h"
+
+#include <iostream>
+#include "easylogging++.h"
+
+textureshaderclass::textureshaderclass()
+{
+	m_vertexShader = 0;
+	m_pixelShader = 0;
+	m_layout = 0;
+	m_matrixBuffer = 0;
+
+	mpSampleState = 0;
+}
+
+
+textureshaderclass::~textureshaderclass()
+{
+}
+
+bool textureshaderclass::Initialize(ID3D11Device* device, HWND hwnd)
+{
+	bool result;
+
+
+	// Initialize the vertex and pixel shaders.
+	result = InitializeShader(device, hwnd, L"texture.vs");
+	if (!result)
+	{
+		return false;
+	}
+
+
+	return true;
+}
+
+void textureshaderclass::Shutdown()
+{
+	// Shutdown the vertex and pixel shaders as well as the related objects.
+	ShutdownShader();
+
+	return;
+}
+
+
+void textureshaderclass::ShutdownShader()
+{
+
+	// Release the sampler state.
+	if (mpSampleState)
+	{
+		mpSampleState->Release();
+		mpSampleState = 0;
+	}
+
+	// Release the matrix constant buffer.
+	if (m_matrixBuffer)
+	{
+		m_matrixBuffer->Release();
+		m_matrixBuffer = 0;
+	}
+
+	// Release the layout.
+	if (m_layout)
+	{
+		m_layout->Release();
+		m_layout = 0;
+	}
+
+	// Release the pixel shader.
+	if (m_pixelShader)
+	{
+		m_pixelShader->Release();
+		m_pixelShader = 0;
+	}
+
+	// Release the vertex shader.
+	if (m_vertexShader)
+	{
+		m_vertexShader->Release();
+		m_vertexShader = 0;
+	}
+
+	return;
+}
+
+
+bool textureshaderclass::Render(ID3D11DeviceContext* deviceContext, int indexCount, XMMATRIX& worldMatrix, XMMATRIX& viewMatrix,
```

Comment:
apDeviceContext, aIndexCount, aWorldMatric, aViewMatrix, ...

I know it's boring but it's not clean right now :(